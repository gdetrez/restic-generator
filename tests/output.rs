use assert_cmd::prelude::*; // Add methods on commands
use std::{error::Error, fs::read_to_string, process::Command};
use tempfile::TempDir;

macro_rules! snapshot_test {
    ($name:ident, $config:expr, $output:expr) => {
        #[test]
        fn $name() -> Result<(), Box<dyn Error>> {
            let normal_dir = TempDir::new()?;
            let early_dir = TempDir::new()?;
            let late_dir = TempDir::new()?;
            let mut cmd = Command::cargo_bin("systemd-restic-generator")?;
            cmd.arg("-c")
                .arg($config)
                .arg(normal_dir.path())
                .arg(early_dir.path())
                .arg(late_dir.path());
            cmd.assert().success();
            insta::assert_snapshot!(read_to_string(normal_dir.path().join($output))?);
            Ok(())
        }
    };
}

snapshot_test!(
    local_backup_service,
    "example-config.toml",
    "restic-myrepo-backup.service"
);

snapshot_test!(
    remote_backup_service,
    "example-config.toml",
    "restic-sftprepo-backup.service"
);

snapshot_test!(
    s3_backup_service,
    "example-config.toml",
    "restic-s3bucket-backup.service"
);
