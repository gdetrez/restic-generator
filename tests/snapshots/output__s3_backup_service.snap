---
source: tests/output.rs
expression: "read_to_string(normal_dir.path().join(\"restic-s3bucket-backup.service\"))?"
---
# generated by restic-generator
[Unit]
Description=backup / to s3:storage.example.com/bucket
SourcePath=example-config.toml
ConditionPathExists=/

[Service]
Environment=RESTIC_REPOSITORY="s3:storage.example.com/bucket"
Environment=AWS_ACCESS_KEY="ABCDEF"
Environment=AWS_SECRET_ACCESS_KEY="s3cr3ts"
Type=oneshot
ExecStartPre=restic unlock
ExecStart=restic backup --host="laptop" --exclude="/home/*/.cache" --exclude="*~" /
SuccessExitStatus=3
Nice=10
IOSchedulingClass=idle

